File: BackUpForABook/UIControls/NGClasses/ANGraphNodeLayer.m
//
//  ANGraphNodeLayer.m
//  AnnieVisualyticsPageView
//
//  Created by hui wang on 1/31/16.
//  Copyright (c) 2016 AnnieLyticx. All rights reserved.
//

#import "ANGraphNodeLayer.h"

@implementation ANGraphNodeLayer

-(instancetype)init
{
    self = [super init];
    
    _node = nil;
    _layer = nil;
    
    return self;
}

-(ANGraphNodeLayer*)initWith:(ANLayoutNode*)node
{
    self = [super init];
    
    _node = node;
    _layer = [[CALayer alloc] init];
    
    return self;
}

// initalize with node and drawing layer
-(ANGraphNodeLayer*)initWith:(ANLayoutNode*)node layer:(CALayer*) layer
{
    self = [super init];
    
    _node = node;
    _layer = layer;
    
    return self;
}

@end

File: Demo_testUI/Demo_testUI/NGClasses_swift/ANGraphNodeLayer.swift
//
//  ANGraphNodeLayer.swift
//  NGClasses
//
//  Created by iMac on 16/12/14.
//  Copyright © 2016年 LyTsai. All rights reserved.
//

import Foundation
import UIKit

// object for holding visual presentation information:
// node - reference to object that has size and location computed
// layer - reference to object that provide visual presentation for this node
class ANGraphNodeLayer {
    // node for data
    var node = ANLayoutNode()
    
    // CALayer for node presentation
    var layer = ANNodeLayer()
    
    // Designated init
    // initalize with node only
    init(_ node: ANLayoutNode){
        self.node = node
        self.layer = ANNodeLayer()
    }
    
    // initalize with node and drawing layer
    init(_ node: ANLayoutNode, layer: CALayer) {
        self.node = node
        self.layer = layer as! ANNodeLayer
    }
}
